### PR TITLE
Association

### PR DESCRIPTION
**Pull request by V0pe**
### Association, Aggregation and Composition

In this project, I did the following to achieve the connection between the classes. 

- [ ] Add the has relationship for `one-many` and `many-many` connection
- [ ] Add the belongs to the relationship
- [ ] Modify the constructor for the relationship
- [ ] Modify the initialization. 
- [ ] Add the method

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,21 +1,24 @@
+require './rentals'
+
 class Nameable
   def correct_name
     raise NotImplementedError
   end
 end
```
Comment (hiromataba) [inline person.rb:8]: - [ ] You should add a class Classroom and add all the associations

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 require './person'
+require './classroom'
```
Comment (hiromataba) [inline student.rb:3]: - [ ] You are currently missing the Book class please add it and all the associations

File: person.rb [hunk 3]
```diff
@@ -1,21 +1,24 @@
+require './rentals'
+
```
Comment (hiromataba) [inline person.rb:2]: - [ ] Please add the Rental class with its associations because it looks like you forgot it

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @V0pe     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (V0pe) [reply person.rb:2]: @hiromataba the rental class have been added with the add rentals method.

File: person.rb (hunk 1 above)
Comment (V0pe) [reply person.rb:8]: @hiromataba A classroom is linked to the student only and not to the person. Could please show the connection

File: student.rb (hunk 2 above)
Comment (V0pe) [reply student.rb:3]: @hiromataba the book class is added to the person. Could you please show the connection?

Comment (hiromataba) [reply student.rb:3]: - [ ] @V0pe  Where is your book class please ?

File: person.rb (hunk 3 above)
Comment (hiromataba) [reply person.rb:2]: - [ ] @V0pe  I cannot see those classes in your files, where are they located please ?


image

File: student.rb (hunk 2 above)
Comment (V0pe) [reply student.rb:3]: @hiromataba the book class is linked straight to the person class. The student inherited the add_rentals from the person class

File: person.rb (hunk 3 above)
Comment (V0pe) [reply person.rb:2]: @hiromataba Apologies, It wasnt saved
Can you go over it again now

Comment (hiromataba) [reply person.rb:2]: @V0pe  Sorry but I cannot check that now because I have already submitted my review but the next reviewer can see it I believe

Comment (V0pe) [reply person.rb:2]: @hiromataba can i call, i have some questions

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require './rentals'
+
+class Bo
+  attr_reader :rentals
```
Comment (medaminedev66) [inline book.rb]: Accordance to (A book has many rentals),
- [ ] Setters for @rentals should be present.

File: rentals.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require './person'
+require './book'
+
+class Rentals
+  attr_reader :person, :book
```
Comment (medaminedev66) [inline rentals.rb]: Accordance to (A rental belongs to a book and also belongs to a person):
- [ ] Setters for @book and @person instance variables should be present

File: person.rb [hunk 6]
```diff
@@ -1,21 +1,24 @@
...
 class Nameable
   def correct_name
     raise NotImplementedError
   end
 end
 
 class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
```
Comment (medaminedev66) [inline person.rb]: Accordance to: a person has many rentals,
- [ ] Setters for `@rentals` should be present.

File: person.rb (hunk 3 above)
Comment (hiromataba) [reply person.rb:2]: @V0pe  I wish it could be possible but I am currently reviewing another project now

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @V0pe 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- The repo looks professional 
- There are no linter errors 
- You have a descriptive PR 
- The associations between classes have been implemented in a good way.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 6 above)
Comment (V0pe) [reply person.rb]: @medaminedev66 Okay.. added

**Review by dasileker — APPROVED** [review]
### **Hi @V0pe,** 

### **Highlights**

Good job on adding all the necessary changes required by CR. 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by V0pe** [comment]
@dasileker Can I interact with you? Thanks.

**Comment by dasileker** [comment]
> @dasileker Can I interact with you? Thanks. 

Anytime on slack

**Comment by V0pe** [comment]
@dasileker Kindly provide your slack

**Pull request merged**